src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,823 @@
+template <typename real2>
+std::array<real2,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convert_conservative_to_primitive_turbulence_model (
+    const std::array<real2,nstate> &conservative_soln) const
+{   
+    std::array<real2,nstate-(dim+2)> primitive_soln_turbulence_model;
+    for(int i=0; i<nstate-(dim+2); i++){
+        primitive_soln_turbulence_model[i] = conservative_soln[dim+2+i]/conservative_soln[0];
+    }
+ 
+    return primitive_soln_turbulence_model;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template <typename real2>
+std::array<dealii::Tensor<1,dim,real2>,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convert_conservative_gradient_to_primitive_gradient_turbulence_model (
+    const std::array<real2,nstate> &conservative_soln,
+    const std::array<dealii::Tensor<1,dim,real2>,nstate> &solution_gradient) const
+{   
+    const std::array<real2,nstate-(dim+2)> primitive_soln_turbulence_model = this->convert_conservative_to_primitive_turbulence_model(conservative_soln); 
+    std::array<dealii::Tensor<1,dim,real2>,nstate-(dim+2)> primitive_soln_gradient_turbulence_model;
+
+    for(int i=0; i<nstate-(dim+2); i++){
+        for(int j=0; j<dim; j++){
+            primitive_soln_gradient_turbulence_model[i][j] = (solution_gradient[dim+2+i][j]-primitive_soln_turbulence_model[i]*solution_gradient[0][j])/conservative_soln[0];
+        }
+    }
+ 
+    return primitive_soln_gradient_turbulence_model;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<dealii::Tensor<1,dim,real>,nstate> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convective_numerical_split_flux(const std::array<real,nstate> &conservative_soln1,
+                                  const std::array<real,nstate> &conservative_soln2) const
+{
+    std::array<dealii::Tensor<1,dim,real>,nstate> conv_num_split_flux;
+    std::array<real,dim+2> conservative_soln1_rans = extract_rans_conservative_solution(conservative_soln1);
+    std::array<real,dim+2> conservative_soln2_rans = extract_rans_conservative_solution(conservative_soln2);
+
+    const real mean_density = this->navier_stokes_physics->compute_mean_density(conservative_soln1_rans, conservative_soln2_rans);
+    const dealii::Tensor<1,dim,real> mean_velocities = this->navier_stokes_physics->compute_mean_velocities(conservative_soln1_rans,conservative_soln2_rans);
+    const std::array<real,nstate-(dim+2)> mean_turbulence_property = compute_mean_turbulence_property(conservative_soln1, conservative_soln2);
+
+    for (int i=0;i<dim+2;i++){
+        conv_num_split_flux[i] = 0.0;
+    }
+    for (int i=dim+2;i<nstate;i++)
+    {
+        for (int flux_dim = 0; flux_dim < dim; flux_dim++){
+            conv_num_split_flux[i][flux_dim] = mean_density*mean_velocities[flux_dim]*mean_turbulence_property[i-(dim+2)];
```

Comment:
is this from a paper?

Reply:
fixed.
As discussed with Prof.Siva, this part should not be here. I removed any relevant parts in the model base and LES that I added earlier.